resources/fetching/fetchers/azure/assets_enricher.go

```diff
@@ -0,0 +1,88 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package fetchers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/samber/lo"
+
+	"github.com/elastic/cloudbeat/resources/fetching/cycle"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
+	"github.com/elastic/cloudbeat/resources/utils/strings"
+)
+
+type AssetsEnricherAPI interface {
```

Comment:
Any specific reason to append it with `API`? I think `AssetsEnricher` is good enough

Reply:
Indeed, and in general, I agree. I just saw the `API` postfix in some other interfaces in cloudbeat, and I decided to follow the pattern.

Reply:
We usually add the `API` suffix when there is a struct and interface with the same name.

Reply:
I don't think that's the case here, is it @moukoublen ?

Reply:
Cool, I renamed it! 👍